src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -986,8 +983,13 @@ FixedwingPositionControl::handle_setpoint_type(const position_setpoint_s &pos_sp
 
 	const float acc_rad = _npfg.switchDistance(500.0f);
 
-	if (pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_POSITION
-	    || pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
+	const bool approaching_vtol_backtransition = _vehicle_status.is_vtol
+			&& pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_POSITION && _position_setpoint_current_valid
+			&& pos_sp_next.type == position_setpoint_s::SETPOINT_TYPE_LAND && _position_setpoint_next_valid;
+
+
+	// check if we should switch to loiter but only if we are not expecting a backtransition to happen
```

Comment:
@KonradRudin 
> Did i miss something, I thought we need now set the next position setpoint triplet to land, in order to detect, that w can ignore the altitude. IS that in and i missed it?

This is where the check is done

Reply:
That is not what i meant, sorry, i saw that we correctly set the next position setpoint to land. but within the https://github.com/PX4/PX4-Autopilot/blob/5aa2518f8cb0277c6036efbf19f7719d529bc143/src/modules/navigator/mission_block.cpp#L65 function, it still only accept it when both within the horizontal and altitude acceptance. You might need to tweak this function here https://github.com/PX4/PX4-Autopilot/blob/5aa2518f8cb0277c6036efbf19f7719d529bc143/src/modules/navigator/navigator_main.cpp#L1108-L1109
To set altitude acceptance to infinite when  we know that the backtransition is happening.
Thinking more about it and seeing, that in fixed wing case, we actually want to make the opposite, maybe we really need to add vtol_back_transition flag from the mission_item also in the setpoint triplet, so we know, that we can ignore altitude.

Reply:
@KonradRudin Maybe I'm missing something but we do set the altitude acceptance radius to infinity if the mission item has the flag `vtol_back_transition` set to true.
https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/navigator/mission_block.cpp#L357C5-L357C18

Reply:
@KonradRudin I made a quick test and increased the altitude of the mission item when the vehicle is approaching the transition by a value of 1000m. To me everything looks nominal, navigator accepts the mission item despite the huge altitude error. Is this what you meant?

![image](https://github.com/PX4/PX4-Autopilot/assets/7610489/a5f7a703-615b-468c-ab94-25019fe4db7c)

Reply:
ITs fine, i missed the line you pointed out to me, thanks, and sorry!